src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int INCREASE = 2;
+    private int currentCapacity;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        currentCapacity = DEFAULT_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        increaseCapacity();
+        Node<K, V> node = new Node<>(key, value, null);
+        int index = getBucketIndex(key);
+        if (table[index] == null) {
+            table[index] = node;
+            size++;
+            return;
+        }
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = node;
+                size++;
```

Comment:
let's call `return` here as well
it looks like after setting `currentNode.next = node;` you are going to next iteration in while loop and then redundantly replace the value